sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/Filter.java

```diff
@@ -0,0 +1,209 @@
+            String fieldName = filter.getFieldName();
+            if (fieldName.startsWith(EXCEPTION_FIELDNAME_PREFIX)) {
+                filter.setFieldName(fieldName.substring(EXCEPTION_FIELDNAME_PREFIX.length()));
+            }
+        });
+    }
+
+    // To be used when checking telemetry against metric charts filters
+    public static boolean checkMetricFilters(DerivedMetricInfo derivedMetricInfo, TelemetryColumns data) {
+        if (derivedMetricInfo.getFilterGroups().isEmpty()) {
+            // This should never happen - even when a user does not add filter pills to the derived metric,
+            // the filterGroups array should have one filter group with an empty array of filters.
+            return true;
+        }
+
+        // Haven't yet seen any case where there is more than one filter group in a derived metric info.
+        // Just to be safe, handling the multiple filter conjunction group case as an or operation.
+        boolean matched = false;
+        for (FilterConjunctionGroupInfo filterGroup : derivedMetricInfo.getFilterGroups()) {
+            matched = matched || checkFilterConjunctionGroup(filterGroup, data);
+        }
+        return matched;
+    }
+
+    // To be used when checking telemetry against document filters. This also gets reused in the logic for checking metrics
+    // charts filters.
+    public static boolean checkFilterConjunctionGroup(FilterConjunctionGroupInfo filterConjunctionGroupInfo,
+        TelemetryColumns data) {
+        // All of the filters need to match for this to return true (and operation).
+        for (FilterInfo filter : filterConjunctionGroupInfo.getFilters()) {
+            if (!checkFilter(filter, data)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean checkFilter(FilterInfo filter, TelemetryColumns data) {
+        try {
+            if (ANY_FIELD.equals(filter.getFieldName())) {
+                return checkAnyFieldFilter(filter, data);
+            } else if (filter.getFieldName().startsWith(CUSTOM_DIM_FIELDNAME_PREFIX)) {
+                return checkCustomDimFilter(filter, data);
+            } else {
+                Object fieldValue = getFieldValue(data, filter.getFieldName());
+
+                if (filter.getFieldName().equals(KnownRequestColumns.SUCCESS)) {
+                    boolean fieldValueBoolean = (Boolean) fieldValue;
+                    boolean comparand = Boolean.parseBoolean(filter.getComparand().toLowerCase());
+                    if (filter.getPredicate().equals(PredicateType.EQUAL)) {
+                        return fieldValueBoolean == comparand;
+                    } else if (filter.getPredicate().equals(PredicateType.NOT_EQUAL)) {
+                        return fieldValueBoolean != comparand;
+                    }
+                } else if (filter.getFieldName().equals(KnownDependencyColumns.RESULT_CODE)
```

Comment:
The code may be easier to read by moving the duration case to a specific else if.